hpx/plugins/parcelport/mpi/mpi_environment.hpp

```diff
@@ -29,27 +30,38 @@ namespace hpx { namespace util
         static int rank();
         static int size();
 
+        static MPI_Comm& communicator();
+
         static std::string get_processor_name();
 
+        struct scoped_lock
+        {
+            scoped_lock();
+            ~scoped_lock();
+        };
+
+        struct scoped_try_lock
+        {
+            scoped_try_lock();
+            ~scoped_try_lock();
+            bool locked;
+        };
+
+        typedef hpx::lcos::local::spinlock mutex_type;
```

Comment:
Still, why do you need your own lock here? Those global lock() and unlock() functions are just wrong.